### ISSUE TITLE
[OpenGL] Metal Gear Solid - Codec screen glitches

### ISSUE THREAD

**Issue by twinaphex** [libretro/beetle-psx-libretro#381]
The Vulkan and software renderers render this properly, but not OpenGL.

See here a video of how the codec screen is built up -

https://www.libretro.com/wp-content/uploads/2016/12/codec.mp4?_=3

For more information on this, see 'Render pass batching' here - 

https://www.libretro.com/index.php/introducing-vulkan-psx-renderer-for-beetlemednafen-psx/

**Comment by IceLancerSR** [comment]
So , how do u get around the bug using vulkan?
I tried every single option , to no effect. Exact same bug is happening in both OGL and Vulkan
https://imgur.com/a/N0iIluj

**Comment by rz5** [comment]
@IceLancerSR The first post of this thread says that this bug happens ONLY on the OpenGL renderer.
If you're getting the graphical bugs seen on the images you posted, it's because you're using the OpenGL renderer.

If you're using RetroArch, set the video driver to 'vulkan' and in beetle-psx-hw, set the core option called "Renderer (restart)" to 'hardware'.

In the future, do not use github issues to ask for help, do it in the libretro Discord.

**Comment by IceLancerSR** [comment]
@rz5 
Yes i am positively using Vulkan renderer and nvidia gpu. Retroarch is set to vulkan and hw core is set to hardware. I don't know, i must have tried this on and off for a year or so and i keep getting same result. After which i give up.
https://i.imgur.com/kYBbENY.jpg

**Comment by IceLancerSR** [comment]
OK!
I found the problem! its looking good on vulkan!
Problem was , that i was using saved core configs ,which will load vulkan render ,but obviously that wasnt good enough. Once i started retroarch alone and changed to Vulkan all turned ok.
I am not familiar with all the ini files retroarch calls when using core configs but aparently switch has to be made to retroarch first , before core calls it.
Problem solved
